plugins/modules/openshift_route.py

```diff
@@ -0,0 +1,543 @@
+
+description:
+  - Looks up a Service and creates a new Route based on it.
+  - Analogous to `oc expose` and `oc create route` for creating Routes, but does not support creating Services.
+  - For creating Services from other resources, see community.kubernetes.k8s_expose
+
+extends_documentation_fragment:
+  - community.kubernetes.k8s_auth_options
+  - community.kubernetes.k8s_wait_options
+  - community.kubernetes.k8s_state_options
+
+requirements:
+  - "python >= 2.7"
+  - "openshift >= 0.11.0"
+  - "PyYAML >= 3.11"
+
+options:
+  service:
+    description:
+      - The name of the service to expose.
+      - Required when I(state) is not absent.
+    type: str
+    aliases: ['svc']
+  namespace:
+    description:
+      - The namespace of the resource being targeted.
+      - The Route will be created in this namespace as well.
+    required: yes
+    type: str
+  labels:
+    description:
+      - Specify the labels to apply to the created Route.
+      - 'A set of key: value pairs.'
+    type: dict
+  name:
+    description:
+      - The desired name of the Route to be created.
+      - Defaults to the value of I(service)
+    type: str
+  hostname:
+    description:
+      - The hostname for the Route.
+    type: str
+  path:
+    description:
+      - The path for the Route
+    type: str
+  wildcard_policy:
+    description:
+      - The wildcard policy for the hostname.
+      - Currently only Subdomain is supported.
+      - If not provided, the default of None will be used.
+    choices:
+      - Subdomain
+    type: str
+  port:
+    description:
+      - Name or number of the port the Route will route traffic to.
+    type: str
+  tls:
+    description:
+      - TLS configuration for the newly created route.
+      - Only used when I(termination) is set.
+    type: dict
+    suboptions:
+      ca_certificate:
+        description:
+          - Path to a CA certificate file on the target host.
+          - Not supported when I(termination) is set to passthrough.
+        type: str
+      certificate:
+        description:
+          - Path to a certificate file on the target host.
+          - Not supported when I(termination) is set to passthrough.
+        type: str
+      destination_ca_certificate:
+        description:
+          - Path to a CA certificate file used for securing the connection.
+          - Only used when I(termination) is set to reencrypt.
+          - Defaults to the Service CA.
+        type: str
+      key:
+        description:
+          - Path to a key file on the target host.
+          - Not supported when I(termination) is set to passthrough.
+        type: str
+      insecure_policy:
+        description:
+          - Sets the InsecureEdgeTerminationPolicy for the Route.
+          - Not supported when I(termination) is set to reencrypt.
+          - When I(termination) is set to passthrough, only redirect is supported.
+          - If not provided, insecure traffic will be disallowed.
```

Comment:
`insecure_policy` should have a default of "disallow" or this line should be removed.